### PR TITLE
Started to vet the documentation and some minor refactors inbetween

### PR DESCRIPTION
**Pull request by Skarlso**

### CODE REVIEW

File: core/empty_response.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package core
 
-// DummyResponse is an empty struct. The main reason behind this
+// EmptyResponse is an empty struct. The main reason behind this
```
Comment (Skarlso) [inline core/empty_response.go:3]: If this is an empty response and it's being used to represent an empty response then let's call it an EmptyResponse. :)

Comment (yitsushi) [reply core/empty_response.go:3]: Agree, originally I used this only to see what's the real response from the server. But true, it should be Empty as we use for 204 responses too.

File: core/empty_response.go [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
 // dummy response is that there are endpoints where the return
```
Comment (yitsushi) [inline core/empty_response.go:4]:
```suggestion
// empty response is that there are endpoints where the return
```

File: core/error.go [hunk 3]
```diff
@@ -6,27 +6,27 @@ import (
 )
 
 const (
-	// ResponseReadError occues when error happened while we tried to
+	// ResponseReadError occurs when error happened while we tried to
```
Comment (yitsushi) [inline core/error.go:9]: How could this pass on lint check as it checks for spelling errors in comments too. 😮

File: core/error.go [hunk 4]
```diff
@@ -72,7 +72,7 @@ func (e InvalidFieldReferenceError) Error() string {
 	)
 }
 
-// RequestValidationError occues when we one of more
```
Comment (yitsushi) [inline core/error.go:75]: Holy, this literally hurts my eyes :embarrassed:

File: core/type.go [hunk 5]
```diff
@@ -1,15 +1,15 @@
 package core
 
-// String is simple string that can be null.
+// String is simple string that can be nil.
```
Comment (yitsushi) [inline core/type.go:3]:
```suggestion
// String is a simple string that can be nil.
```

**Pull request merged**